File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/specialtypes/interfaces/vb/Interfaces.vb

Imports System

Class Point : Implements IComparable

    Public x, y As Int32
    
    Public Sub New(x As Int32, y As Int32)
        Me.x = x
        Me.y = y
    End Sub 'New
    
    
    ' CompareTo is define by the IComparable interface
    Public Function CompareTo(other As Object) As Integer Implements IComparable.CompareTo
        Dim p As Point = CType(other, Point)
        
        Dim thisDistanceFromOrigin As Double = Math.Sqrt((x * x + y * y))
        Dim otherDistanceFromOrigin As Double = Math.Sqrt((p.x * p.x + p.y * p.y))
        
        Return Math.Sign((thisDistanceFromOrigin - otherDistanceFromOrigin))
    End Function 'CompareTo
    
    
    Public Overrides Function ToString() As String
        Return String.Format("({0}, {1})", x, y)
    End Function 'ToString
End Class 'Point



Class App
    Shared Sub Main()
        Dim points(4) As Point
        points(0) = New Point(2, 2)
        points(1) = New Point(3, 2)
        points(2) = New Point(2, 3)
        points(3) = New Point(7, 8)
        points(4) = New Point(0, 1)
        
        Array.Sort(points)
        
        ' Display all the elements in the array
        Dim i As Int32
        For i = 0 To points.Length - 1
            Console.WriteLine("Point {0}: {1}", i, points(i))
        Next i

        Console.Write("Press Enter to close window...")
        Console.Read()
    End Sub 'Main
End Class 'App

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/specialtypes/interfaces/cs/Interfaces.cs
using System;

class Point : IComparable {
   public Int32 x, y;

   public Point(Int32 x, Int32 y) {
      this.x = x;
      this.y = y;
   }

   // CompareTo is define by the IComparable interface
   public Int32 CompareTo(Object other) {
      Point p = (Point) other;

      Double thisDistanceFromOrigin = Math.Sqrt(x * x + y * y);
      Double otherDistanceFromOrigin = Math.Sqrt(p.x * p.x + p.y * p.y);

      return(Math.Sign(thisDistanceFromOrigin - otherDistanceFromOrigin));
   }

   public override String ToString() {
      return(String.Format("({0}, {1})", x, y));
   }
}


class App {
   static void Main() {
      Point[] points = new Point[5];
      points[0] = new Point(2, 2);
      points[1] = new Point(3, 2);
      points[2] = new Point(2, 3);
      points[3] = new Point(7, 8);
      points[4] = new Point(0, 1);

      Array.Sort(points);

      // Display all the elements in the array
      for (Int32 i = 0; i < points.Length; i++) 
         Console.WriteLine("Point {0}: {1}", i, points[i]);

      Console.Write("Press Enter to close window...");
      Console.Read();
   }
}